Source/SwiftLintBuiltInRules/Rules/Idiomatic/NoGroupingExtensionRule.swift

```diff
@@ -22,38 +23,100 @@ struct NoGroupingExtensionRule: OptInRule {
     )
 
     func validate(file: SwiftLintFile) -> [StyleViolation] {
-        let collector = NamespaceCollector(dictionary: file.structureDictionary)
-        let elements = collector.findAllElements(of: [.class, .enum, .struct, .extension])
+        return Visitor(configuration: configuration, file: file)
```

Comment:
Totally up to you but I find that using implicit returns and if-expressions from Swift 5.9 to make this more readable:

```swift
Visitor(configuration: configuration, file: file)
    .walk(file: file) { visitor in
        visitor.extensionDeclarations.compactMap { decl in
            if visitor.typeDeclarations.contains(decl.name) {
                ReasonedRuleViolation(position: decl.position)
            } else {
                nil
            }
        }
    }
    .sorted()
    .map { makeViolation(file: file, violation: $0) }
```